docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,296 @@
+
+!!! note 
+    The port numbers for all EdgeX services have changed which must be reflected in the `Clients` configuration.
+
+!!! example "Example - Core Metadata client configuration"
+    ```toml
+      [Clients]
+        [Clients.core-metadata]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59881
+    ```
+
+!!! example "Example - All available clients configured with new port numbers"
+    ```toml
+      [Clients]
+        # Used for version check on start-up
+        # Also used for DeviceSevice, DeviceProfile and Device clients
+        [Clients.core-metadata]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59881
+        
+        # Used for Event client which is used by PushToCoreData function
+        [Clients.core-data]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59880
+        
+        # Used for Command client
+        [Clients.core-command]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59882
+        
+        # Used for Notification and Subscription clients
+        [Clients.support-notifications]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59860
+    ```
+
+#### Trigger
+
+The `Trigger` section (previously named `Binding`) has been restructured with `EdgexMessageBus` (previously named `MessageBus`) and `ExternalMqtt` (previously named `MqttBroker` ) moved under it. The `SubscribeTopics` (previously named `SubscribeTopic`) has been moved under the `EdgexMessageBus.SubscribeHost` and `ExternalMqtt` sections. The `PublishTopic` has been moved under the `EdgexMessageBus.PublishHost` and `ExternalMqtt` sections.
+
+##### EdgeX MessageBus
+
+If your Application Service is using the EdgeX MessageBus trigger, you can simply copy the complete `Trigger` configuration from the example below and tweak it as needed. 
+
+!!! example "Example - EdgeX MessageBus trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="edgex-messagebus"
+      [Trigger.EdgexMessageBus]
+      Type = "redis"
+        [Trigger.EdgexMessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        SubscribeTopics="edgex/events/#"
+        [Trigger.EdgexMessageBus.PublishHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        PublishTopic="example"
+        [Trigger.EdgexMessageBus.Optional]
+        AuthMode = 'usernamepassword'  # required for redis messagebus (secure or insecure).
+        SecretName = 'redisdb'
+    ```
+
+From the above example you can see the improved structure and the following changes:
+
+- Default `EdgexMessageBus` type has changed to `Redis` from `ZeroMQ`
+- Type value for `Redis` has changed from `redistreams` to `redis`. This is because the implementation no longer uses Redis Streams. It now using Redis Pub/Sub.
```

Comment:
also pick either `from ... to ...` or `to ... from ...`

In the first of the changes bulleted, you use `to Redis from ZeroMQ`, and in the following change bulleted you put `from redistreams to redis`. This is nit-picky I know :), but it would be nicer if you were consistent here.

Reply:
Nice catch! Will fix.